build/update-match-schema.sh

```diff
@@ -0,0 +1,38 @@
+# The purpose of the script is to build a golang string constant containing the
```

Comment:
is there a test that we could come up with to make sure that the behavior of this script doesn't diverge down the road? I do see the panics in `init()` for `target` but I am wondering if that's the best way to go 🤔 

Or are the bats tests passing the actual litmus test that the file gets generated properly?

Reply:
If controller-gen were to change the way CRDs are generated in a way that "hard" breaks this script, it is very likely we wouldn't even be able to deserialize the yaml string into a struct, in which case the `panic` in `init()` would get triggered. 

If we still were to generate valid yaml despite a controller-gen change that "soft" breaks this script (e.g. the right fields were not parsed out, but the yaml is still valid), some, if not all, of the various unit tests that create `Constraint` or `Mutation` kinds from yaml should fail. 

I suppose we could add another test that explicitly tries to create a Constraint/Mutator containing `Match` from yaml, but I would argue this is redundant since we already have tests that do that. `.pkg/gator/verify` is one such example.